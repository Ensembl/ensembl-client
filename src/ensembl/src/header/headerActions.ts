import { createAction, createStandardAction } from 'typesafe-actions';
import analyticsTracking from 'src/services/analytics-service';

export const toggleAccount = createStandardAction('header/toggle-account')();

export const toggleLaunchbar = createStandardAction(
  'header/toggle-launchbar'
)();

export const changeCurrentApp = createAction(
  'header/change-current-app',
  (resolve) => {
    return (appName: string) => {
      analyticsTracking.setPageDimension(appName);
      return resolve(appName);
    };
  }
);
