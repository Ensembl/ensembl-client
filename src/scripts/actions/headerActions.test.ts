import configureStore from 'redux-mock-store';

import * as header from './headerActions';

const mockStore = configureStore();
const store = mockStore();

describe('headerActions', () => {
  beforeEach(() => {
    store.clearActions();
  });

  test('dispatches toggle actions correctly', () => {
    const expectedActions = [
      {
        type: 'header/toggle-account'
      },
      {
        type: 'header/toggle-launchbar'
      }
    ];

    store.dispatch(header.toggleAccount());
    store.dispatch(header.toggleLaunchbar());

    expect(store.getActions()).toEqual(expectedActions);
  });

  test('dispatches changeCurrentApp action correctly', () => {
    const expectedActions = [
      {
        payload: 'global-search',
        type: 'header/change-current-app'
      }
    ];

    store.dispatch(header.changeCurrentApp('global-search'));
    expect(store.getActions()).toEqual(expectedActions);
  });

  test('dispatches the correct actions and payloads', () => {
    store.dispatch(header.toggleAccount());
    store.dispatch(header.toggleLaunchbar());
    store.dispatch(header.changeCurrentApp('global-search'));

    expect(store.getActions()).toMatchSnapshot();
  });
});
