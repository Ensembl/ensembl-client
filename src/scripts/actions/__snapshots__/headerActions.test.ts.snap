// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`headerActions dispatches the correct actions and payloads 1`] = `
Array [
  Object {
    "meta": undefined,
    "payload": undefined,
    "type": "header/toggle-account",
  },
  Object {
    "meta": undefined,
    "payload": undefined,
    "type": "header/toggle-launchbar",
  },
  Object {
    "meta": undefined,
    "payload": "global-search",
    "type": "header/change-current-app",
  },
]
`;
